kie-camel/src/test/java/org/kie/camel/embedded/camel/component/KieEmbeddedEndpointDMNTest.java

```diff
@@ -0,0 +1,120 @@
+ * limitations under the License.
+ */
+
+package org.kie.camel.embedded.camel.component;
+
+import java.util.List;
+
+import javax.naming.Context;
+import javax.naming.NamingException;
+
+import org.apache.camel.Processor;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.model.ClaimCheckOperation;
+import org.drools.compiler.kie.builder.impl.DrlProject;
+import org.junit.Test;
+import org.kie.api.KieServices;
+import org.kie.api.builder.KieBuilder;
+import org.kie.api.builder.KieFileSystem;
+import org.kie.api.builder.Message;
+import org.kie.api.io.KieResources;
+import org.kie.api.runtime.KieSession;
+import org.kie.camel.embedded.dmn.DecisionsToHeadersProcessor;
+import org.kie.camel.embedded.dmn.ToDMNEvaluateAllCommandProcessor;
+import org.kie.camel.embedded.dmn.ToMapProcessor;
+import org.kie.pipeline.camel.Person;
+
+public class KieEmbeddedEndpointDMNTest extends KieCamelTestSupport {
+
+    @Test
+    public void testDMN() throws Exception {
+        Person johnDoe = new Person("John Doe", 47);
+        Person alice = new Person("Alice", 18);
+        Person bob = new Person("Bob", 17);
+        Person charlie = new Person("Charlie", 21);
+
+        MockEndpoint canDrink = getMockEndpoint("mock:canDrink");
+        canDrink.expectedMessageCount(3);
+        canDrink.expectedBodiesReceived(johnDoe, alice, charlie);
+
+        MockEndpoint noDrink = getMockEndpoint("mock:noDrink");
+        noDrink.expectedMessageCount(1);
+        noDrink.expectedBodiesReceived(bob);
+
+        template.requestBody("direct:start", johnDoe);
+        template.requestBody("direct:start", alice);
+        template.requestBody("direct:start", bob);
+        template.requestBody("direct:start", charlie);
+
+        canDrink.assertIsSatisfied();
+        noDrink.assertIsSatisfied();
+    }
+
+    @Override
+    protected RouteBuilder createRouteBuilder() throws Exception {
+        final Processor toMap = new ToMapProcessor("a person");
+        final Processor toDMNCommand = new ToDMNEvaluateAllCommandProcessor("https://kiegroup.org/dmn/_FD3D17D0-D23E-457E-B41A-380644F030A8",
+                                                                            "Can Drink?",
+                                                                            "dmnResult");
+        final Processor dmnToHeader = DecisionsToHeadersProcessor.builder("dmnResult",
+                                                                          "canDrinkHeader",
+                                                                          "Can Drink?")
+                                                                 .build();
+
+        return new RouteBuilder() {
+            public void configure() throws Exception {
+                from("direct:start").claimCheck(ClaimCheckOperation.Push)
+                                    .process(toMap)
+                                    .process(toDMNCommand)
```

Comment:
What about merge `ToMapProcessor` logic with `ToDMNEvaluateAllCommandProcessor`? Something like an additional optional parameter for the constructor to specify where to put the exchange body. If this value is not provided and the body is a `Map` we can use the whole object as input.
Wdyt?

Reply:
I thought about that indeed, but at least in my experience Camel route is very similar to ETL, so you want the basic components/block to not do "too much" if possible.
This way:

- if the user want to collapse the two steps into one, can always do it since they are very trivial
- if the user has other steps or different steps, can compose the steps in the best fit: example, the user might be looking to transform the exchange into a Map as part of another ETL step.

So I see your point, but providing the basic steps and the basic version of the components, is the best approach in my limited Camel experience